lib/element.js

```diff
@@ -161,6 +161,100 @@ ElementArrayFinder.prototype.all = function(locator) {
 };
 
 /**
+ * Get a slice of elements within the ElementArrayFinder.
+ * Negative indices are wrapped (i.e. -i means ith element from last)
+ * This does not actually retrieve the underlying element.
+ *
+ * Slice can be negative in which case we go backwards.
+ *
+ * Note that if step is positive and start > stop, an error is thrown
+ * Note that if step is negative and start < stop, an error is thrown
+ * Note that slice(null,null,-1) will NOT produce a reversed array - use reverse() on the result instead.
+ *
+ * @view
+ * <ul class="items">
+ *   <li>Zeroth</li>
+ *   <li>First</li>
+ *   <li>Second</li>
+ *   <li>Third</li>
+ *   <li>Fourth</li>
+ *   <li>Fifth</li>
+ *   <li>Sixth</li>
+ *   <li>Seventh</li>
+ * </ul>
+ *
+ * @example
+ * var list = element.all(by.css('.items li'));
+ *
+ * expect(element.all(locator).slice(1,4).getText()).toEqual(['First','Second','Third']);
+ * expect(element.all(locator).slice(6).getText()).toEqual(['Sixth','Seventh']);
+ * expect(element.all(locator).slice(-1,-5,-2).getText()).toEqual(['Seventh','Fifth']);
+ *
+ * @param {sliceStart} start of the slice. Defaults to 0.
+ * @param {sliceStop} end of the slice(not included). Defaults to last element.
+ * @param {sliceStep} The jump. Defaults to 1.
+ * @return {ElementArrayFinder} finder representing sliced elements
+ */
+ElementArrayFinder.prototype.slice = function (sliceStart,sliceStop,sliceStep) {
+  var self = this;
+
+  var getWebElements = function () {
```

Comment:
It might be easier to implement this on top of [ElementArrayFinder.prototype.filter](http://www.protractortest.org/#/api?view=ElementArrayFinder.prototype.filter)